scrapy_poet/downloadermiddlewares.py

```diff
@@ -148,9 +148,11 @@ def process_response(
         )
         # Fill the callback arguments with the created instances
         for arg, value in final_kwargs.items():
-            # Precedence of user callback arguments
-            if arg not in request.cb_kwargs:
-                request.cb_kwargs[arg] = value
+            # If scrapy-poet can't provided the dependency, allow the user to
+            # give it.
+            if value is None and request.cb_kwargs.get(arg) is not None:
```

Comment:
I think this is a correct implementation, but maybe `arg in request.cb_kwargs` is simpler?

Reply:
Updated in https://github.com/scrapinghub/scrapy-poet/pull/113/commits/af6905a01f6ba6fe008c35d37fcc5889fdc39443